tests/nosetests/pyanaconda_tests/module_source_cdrom_test.py

```diff
@@ -0,0 +1,310 @@
+        # The other two tests are needed only to test the exit when nothing is found.
+
+        device_tree = self.set_up_device_tree(4)
+        proxy_getter.return_value = device_tree
+
+        mount_mock.side_effect = \
+            [MountFilesystemError("Mocked failure"), DEFAULT, DEFAULT, DEFAULT]
+
+        # only for devices 2-4; the expected first call is prevented by the exception from mount
+        valid_mock.side_effect = [False, True, False]
+
+        task = SetUpCdromSourceTask("/mnt/put-cdrom-here")
+        task.run()
+
+        # 3/4 devices tried, 1/4 untried
+        self.assert_resolve_and_mount_calls(device_tree, mount_mock, 3, 1)
+
+        # Only 2 & 3 were mounted
+        self.assertEqual(valid_mock.call_count, 2)
+        # It makes no sense to check how validation was called because all mounting is to the same
+        # path.
+
+        #  #1 died earlier, #2 was unmounted, #3 was left mounted, #4 never got mounted
+        unmount_mock.assert_called_once_with("/mnt/put-cdrom-here")
+        self.assertEqual(task._device, "test2")
+
+        # Task won't run twice
+        with self.assertRaises(SourceSetupError) as cm:
+            task.run()
+        self.assertEqual(str(cm.exception), "CD-ROM already mounted")
+
+    @patch("pyanaconda.modules.payloads.source.cdrom.initialization._is_valid_install_disk")
+    @patch("pyanaconda.modules.payloads.source.cdrom.initialization.unmount")
+    @patch("pyanaconda.modules.payloads.source.cdrom.initialization.mount")
+    @patch_dbus_get_proxy
+    def run_fail_mount_test(self, proxy_getter, mount_mock, unmount_mock, valid_mock):
+        """Test CD-ROM Source setup installation source task run - mount failure."""
+        # One disk, fails to mount, expect exception
+
+        device_tree = self.set_up_device_tree(1)
+        proxy_getter.return_value = device_tree
+
+        mount_mock.side_effect = MountFilesystemError("Mocked failure")
+        valid_mock.return_value = True
+
+        with self.assertRaises(SourceSetupError) as cm:
+            task = SetUpCdromSourceTask("/mnt/put-cdrom-here")
+            task.run()
+
+        # 1/1 devices tried, 0 untried
+        self.assert_resolve_and_mount_calls(device_tree, mount_mock, 1, 0)
+        # neither validation nor unmounting could not have been reached
+        valid_mock.assert_not_called()
+        unmount_mock.assert_not_called()
+        # no device was selected
+        self.assertFalse(task._device)
+        # exception happened due to no disk
+        self.assertEqual(str(cm.exception), "Found no CD-ROM")
+
+    @patch("pyanaconda.modules.payloads.source.cdrom.initialization._is_valid_install_disk")
+    @patch("pyanaconda.modules.payloads.source.cdrom.initialization.unmount")
+    @patch("pyanaconda.modules.payloads.source.cdrom.initialization.mount")
+    @patch_dbus_get_proxy
+    def run_no_good_test(self, proxy_getter, mount_mock, unmount_mock, valid_mock):
```

Comment:
Why do you have `run_` prefix for these tests? Also `run_no_good` doesn't ring a bell for me about what is tested here.

Reply:
Changed...